merlin/models/torch/utils/schema_utils.py

```diff
@@ -0,0 +1,98 @@
+import torch
+
+from merlin.schema import ColumnSchema, Schema, Tags
+
+
+class SchemaTrackingMixin:
+    """
+    A mixin class for PyTorch modules to track the output shapes and dtypes
+    of the forward pass. This is used in order to automatically generate
+    the output-schema.
+
+    It registers a hook to capture this information and
+    provides methods to access the output schema, as well as to set the module
+    in training or evaluation mode.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._register_schema_tracking_hook()
+
+    def _post_forward_hook(self, module, input, output):
+        """Hook function to be called after the forward pass of the module.
+
+        Parameters
+        ----------
+        module : torch.nn.Module
+            The module for which the forward pass was called.
+        input : tuple
+            The input arguments passed to the forward method.
+        output : torch.Tensor or dict
+            The output of the forward method.
+        """
+        if not module._forward_called:
+            if isinstance(output, dict):
+                for key, value in output.items():
+                    module._output_shapes[key] = value.shape
+                    module._output_dtypes[key] = value.dtype
+            else:
+                module._output_shapes["output"] = output.shape
```

Comment:
this `"output"` key is for the case where a module outputs a single tensor? Would something that uses this output schema depend on the name of this column name elsewhere?